design/ceph-csi-driver.md

```diff
@@ -41,18 +50,93 @@ Rook Ceph Agent supports Ceph mon failover by storing Ceph cluster instead of Ce
+With the addition of CSI 1.0 support in Kubernetes 1.13 Rook should become a
+fully-featured method of deploying Ceph-CSI aiming to minimize extra steps
+needed to use CSI targeting the Rook managed Ceph cluster(s). Initially, this
+would be an opt-in feature that requires Kubernetes 1.13. Supporting CSI
+versions earlier than 1.0 will be a non-goal.
+
+Opting in to CSI should be simple and require very few changes from the
+currently documented approach for deploying Rook. Configuring Rook to use CSI
+should require changing only the default mechanism for interacting with the
+storage classes. The standard deployment should include the needed RBAC files
+for managing storage with Ceph-CSI. Rook should package and/or source all other
+needed configuration files/templates. All other configuration must be defaulted
+to reasonable values and only require changing if the user requires it.
+
+The following are plans to converge the user experience when choosing to use
+CSI rather than the flex volume method:
+
+#### Ceph-CSI Requirements
+
+To manage CSI with Rook the following requirements are placed on the
+ceph-csi project:
+* Configuration parameters currently set in the Storage Class will be
+  configurable via secrets:
+  * Mons (done)
+  * Admin Id
+  * User Id
+* The key "blockpool" will serve as an alias to "pool"
+
+To additionally minimize the required parameters in a Storage Class
+it may require changes to create CSI instance secrets; secrets
+that are associated with CSI outside of the storage class (see https://github.com/ceph/ceph-csi/pull/224).
+If this change is made nearly no parameters will be directly required in the
+storage class.
+
+#### Rook Requirements
+
+To manage CSI with Rook the following requirements are place on Rook:
+* Rook deployments must include all the needed RBAC rules to set up CSI
+* Rook deploys all additional CSI components required to provision
+  and mount a volume
+* Rook must be able to dynamically update the secrets used to configure
+  Ceph-CSI, including but not limited to the mons list.
+* Users should not be required to deploy Rook differently when using
+  CSI versus flex except minimal steps to opt in to CSI
+* When provisioning Ceph-CSI Rook must uniquely identify the
+  driver/provisioner name so that multiple CSI drivers or multiple Rook
+  instances within a (Kubernetes) cluster will not collide
+
+
+### Future points of integration
+
+While not immediately required this section outlines a few improvements
+that could be made to improve the Rook and CSI integration:
+
+#### Extend CephBlockPool and CephFilesystem CRDs to provision Storage Classes
+
+Extend CephBlockPool and CephFilesystem CRDs to automatically provision Storage
```

Comment:
This question has come up before and I'm sure there's a github issue somewhere around this. Seems like the benefit of generating a storage class would be limited. If they're defining the CRD, why wouldn't they just define the storage class as well? It's a separate resource.

Reply:
OK, I didn't find it yet. Since this is a "futures" section are you OK with leaving this here for now as a reminder for more discussion or do we hash it out more in this PR?

Reply:
Sure, we can leave it as tentative